sdl-core/src/test/scala/io/smartdatalake/workflow/action/ScalaClassSparkDsNTo1TransformerTest.scala

```diff
@@ -287,6 +292,47 @@ class ScalaClassSparkDsNTo1TransformerTest extends FunSuite with BeforeAndAfter
     directoriesToDelete.foreach(dir => dir.deleteRecursively())
   }
 
+  test("One Ds2To1 Transformation using config file: one input type that is partioned, using addPartitionValuesToOutput = true") {
+
+    // setup DataObjects
+    val partitionValues = Seq(PartitionValues(
+      Map(("year" -> "1992"),
+        ("month" -> "04"),
+        ("day" -> "25"))))
+    // fill src with first files
+    val srcDO1 = SparkSubFeed(SparkDataFrame(
+      Seq(("john", 5, "1992", "04", "25"))
+        .toDF("name", "rating", "year", "month", "day")
+    ), DataObjectId("src1Ds"), partitionValues)
+
+    val srcDO2 = SparkSubFeed(SparkDataFrame(
+      Seq(("doe", 10, "1992", "04", "25"))
+        .toDF("name", "rating", "year", "month", "day")
+    ), DataObjectId("src2Ds"), partitionValues)
+
+    val sdlb = new DefaultSmartDataLakeBuilder()
+
+    val sdlConfig = SmartDataLakeBuilderConfig(feedSel = "test_feed_name", configuration = Some(Seq(
+      getClass.getResource("/configScalaClassSparkDsNto1Transformer/usingDataObjectIdWithPartitionAutoSelect.conf").getPath)),
+      partitionValues =
+        Some(Seq(PartitionValues(
+          Map(("year" -> "1992"),
+            ("month" -> "04"),
+            ("day" -> "25")
+          ))))
+    )
+    //Run SDLB
+    val (subFeeds, _): (Seq[SubFeed], Map[RuntimeEventState, Int]) = sdlb.startSimulationWithConfigFile(sdlConfig, Seq(srcDO1, srcDO2))(session)
```

Comment:
interestingly, even though startSimulationWithConfigFile should only execute init-phase, I'm able to test it with real data. Now Idea why

Reply:
exec Phase includes writing DataObjects, thats why simulation is using init. But in fact there are several special rules for simulation in init phase... see usage of ActionPipelineContext.simulation.